src/nvim/message.c

```diff
@@ -1450,6 +1450,9 @@ void msg_prt_line(char_u *s, int list)
       } else if (c == ' ' && trail != NULL && s > trail) {
         c = lcs_trail;
         attr = hl_attr(HLF_8);
+      } else if (c == ' ' && list && lcs_space != NUL) {
```

Comment:
Sure thing. I've looked at LuaJIT's, but haven't seen Tcl's. Another one which I'd recommend is [Plan 9's source tree](http://plan9.bell-labs.com/plan9/download/plan9.iso.bz2), which has plenty of novel practices throughout the codebase. That will be all, as I don't want to distract any more from this issue :P